src/GafferImage/Display.cpp

```diff
@@ -175,18 +167,19 @@ class GafferDisplayDriver : public IECoreImage::DisplayDriver
 					for( int channelIndex = 0, numChannels = channelNames().size(); channelIndex < numChannels; ++channelIndex )
 					{
 						const V2i tileOrigin( tileOriginX, tileOriginY );
-						ConstFloatVectorDataPtr tileData = getTile( tileOrigin, channelIndex );
-						if( !tileData )
+						int tileIndex = getTileIndex( tileOrigin, channelIndex );
+						if( tileIndex == -1 )
```

Comment:
How about we hide the indexing completely, and replace `getTileIndex()` with `TileCacheElement *tile()`, returning `nullptr` when there is no tile?